libraries/chain/block_log.cpp

```diff
@@ -462,8 +462,7 @@ namespace eosio { namespace chain {
          inline static uint32_t  default_initial_version = block_log::max_supported_version;
 
          std::mutex       mtx;
-         signed_block_ptr head;
-         block_id_type    head_id;
+         std::optional<std::pair<signed_block_ptr, block_id_type>> head;
```

Comment:
I like using optional. But I am a little hesitate about std::pair, as when you look at head->first or head->second, you need to go back the definition to see what first and second refer to. Can it be made an optional of struct?

Reply:
Done.